packages/components/src/__future__/Tag/BaseTag.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { ReactNode } from "react"
+import classNames from "classnames"
+
+export type TagColors =
+  | "gray"
+  | "blue"
+  | "green"
+  | "yellow"
+  | "orange"
+  | "red"
+  | "purple"
+
+export interface BaseTagProps {
+  children: ReactNode
+  color?: TagColors
+}
+
+export const BaseTag = ({
```

Comment:
Still not sure what the value is of using the `BaseTag` as a consumer creating a custom `Tag` when I can simply use `<Tag><CustomElement /></Tag>` rather than importing a completely separate component?

Wouldn't your Avatar example still work if I did this:
```
<Tag>
    <span className="-ms-4 -my-[1px]">
      <Avatar {...avatarProps} size="small" />
    </span>
    {children}
</Tag>
```

Reply:
Yes, you are right. I got confused when moving the logic around for the styling.